nexus/src/authz/api_resources.rs

```diff
@@ -47,38 +47,36 @@ use uuid::Uuid;
 
 /// Describes an authz resource that corresponds to an API resource that has a
 /// corresponding ResourceType and is stored in the database
-pub trait ApiResource: Clone + Send + Sync + 'static {
-    /// If roles can be assigned to this resource, return the type and id of the
-    /// database record describing this resource
+pub trait ApiResource:
```

Comment:
I found the combination of `ApiResource` and `ApiResourceError` unnecessarily confusing, so I combined them.  This basically meant removing the `Clone + oso::PolarClass` trait bounds here (so that `ApiResource` can be object-safe) and adding them back (actually `oso::ToPolar`) in the few places where we actually need them.